vyper/semantics/validation/utils.py

```diff
@@ -456,9 +459,22 @@ def validate_expected_type(node, expected_type):
             types_str = sorted(str(i) for i in given_types)
             given_str = f"{', '.join(types_str[:1])} or {types_str[-1]}"
 
+        suggestions_str = ""
+        if (
+            len(expected_type) == 1
+            and isinstance(expected_type[0], AddressDefinition)
+            and len(given_types) == 1
+            and isinstance(given_types[0], Bytes20Definition)
```

Comment:
nit -- because of preconditions earlier in this function, we know that both lists are nonzero. i think it's worth it to slightly generalize this check and remove the `len() == 1` checks, because it's easier from a maintenance standpoint to constrain the check if it turns out to be wrong than to relax the constraint.